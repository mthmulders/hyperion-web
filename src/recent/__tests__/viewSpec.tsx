import * as React from 'react';

import { shallow } from 'enzyme';

import View from '../view';

describe('<View />', () => {
    it('should present some data', () => {
        // Arrange
        const data = [
            {
                elecCon: 0.265,
                elecConsLow: 2150.714,
                elecConsNormal: 2702.641,
                elecProd: 0.000,
                elecProdLow: 0.000,
                elecProdNormal: 0.000,
                gas: 3781.676,
                tariff: '0002',
                ts: new Date('2018-01-05T15:40:47+01:00'),
            },
        ];

        // Act
        const wrapper = shallow(<View data={ data } />);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });
});
