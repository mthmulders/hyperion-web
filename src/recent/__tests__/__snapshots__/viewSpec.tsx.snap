// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> should present some data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentView
    data={
      Array [
        Object {
          "elecCon": 0.265,
          "elecConsLow": 2150.714,
          "elecConsNormal": 2702.641,
          "elecProd": 0,
          "elecProdLow": 0,
          "elecProdNormal": 0,
          "gas": 3781.676,
          "tariff": "0002",
          "ts": "2018-01-05T15:40:47+01:00",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "barCategoryGap": "10%",
      "barGap": 4,
      "children": Array [
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />,
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="label"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />,
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          interval={0}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          unit=" W"
          width={60}
          yAxisId={0}
        />,
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "height": 0,
              "width": 0,
              "x": 0,
              "y": 0,
            }
          }
          wrapperStyle={Object {}}
        />,
        <Legend
          align="center"
          iconSize={14}
          layout="horizontal"
          verticalAlign="bottom"
        />,
        <Line
          activeDot={true}
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="value"
          dot={true}
          fill="#fff"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          name="Power consumption"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="#3182bd"
          strokeWidth={1}
          type="monotone"
          xAxisId={0}
          yAxisId={0}
        />,
      ],
      "data": Array [
        Object {
          "label": "15:40:47",
          "value": 265,
        },
      ],
      "height": 350,
      "layout": "horizontal",
      "margin": Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      },
      "reverseStackOrder": false,
      "stackOffset": "none",
      "width": 1200,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fill": "none",
          "horizontal": true,
          "horizontalFill": Array [],
          "horizontalPoints": Array [],
          "stroke": "#ccc",
          "vertical": true,
          "verticalFill": Array [],
          "verticalPoints": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowDataOverflow": false,
          "allowDecimals": true,
          "allowDuplicatedCategory": true,
          "dataKey": "label",
          "domain": Array [
            0,
            "auto",
          ],
          "height": 30,
          "hide": false,
          "mirror": false,
          "orientation": "bottom",
          "padding": Object {
            "left": 0,
            "right": 0,
          },
          "reversed": false,
          "scale": "auto",
          "tickCount": 5,
          "type": "category",
          "width": 0,
          "xAxisId": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowDataOverflow": false,
          "allowDecimals": true,
          "allowDuplicatedCategory": true,
          "domain": Array [
            0,
            "auto",
          ],
          "height": 0,
          "hide": false,
          "interval": 0,
          "mirror": false,
          "orientation": "left",
          "padding": Object {
            "bottom": 0,
            "top": 0,
          },
          "reversed": false,
          "scale": "auto",
          "tickCount": 5,
          "type": "number",
          "unit": " W",
          "width": 60,
          "yAxisId": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "animationDuration": 400,
          "animationEasing": "ease",
          "contentStyle": Object {},
          "coordinate": Object {
            "x": 0,
            "y": 0,
          },
          "cursor": true,
          "cursorStyle": Object {},
          "filterNull": true,
          "isAnimationActive": true,
          "itemSorter": [Function],
          "itemStyle": Object {},
          "labelStyle": Object {},
          "offset": 10,
          "separator": " : ",
          "useTranslate3d": false,
          "viewBox": Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          },
          "wrapperStyle": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "iconSize": 14,
          "layout": "horizontal",
          "verticalAlign": "bottom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeDot": true,
          "animateNewValues": true,
          "animationBegin": 0,
          "animationDuration": 1500,
          "animationEasing": "ease",
          "connectNulls": false,
          "dataKey": "value",
          "dot": true,
          "fill": "#fff",
          "hide": false,
          "isAnimationActive": true,
          "legendType": "line",
          "name": "Power consumption",
          "onAnimationEnd": [Function],
          "onAnimationStart": [Function],
          "points": Array [],
          "stroke": "#3182bd",
          "strokeWidth": 1,
          "type": "monotone",
          "xAxisId": 0,
          "yAxisId": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "barCategoryGap": "10%",
        "barGap": 4,
        "children": Array [
          <CartesianGrid
            fill="none"
            horizontal={true}
            horizontalFill={Array []}
            horizontalPoints={Array []}
            stroke="#ccc"
            vertical={true}
            verticalFill={Array []}
            verticalPoints={Array []}
          />,
          <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="label"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={30}
            hide={false}
            mirror={false}
            orientation="bottom"
            padding={
              Object {
                "left": 0,
                "right": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="category"
            width={0}
            xAxisId={0}
          />,
          <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={0}
            hide={false}
            interval={0}
            mirror={false}
            orientation="left"
            padding={
              Object {
                "bottom": 0,
                "top": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="number"
            unit=" W"
            width={60}
            yAxisId={0}
          />,
          <Tooltip
            active={false}
            animationDuration={400}
            animationEasing="ease"
            contentStyle={Object {}}
            coordinate={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            cursor={true}
            cursorStyle={Object {}}
            filterNull={true}
            isAnimationActive={true}
            itemSorter={[Function]}
            itemStyle={Object {}}
            labelStyle={Object {}}
            offset={10}
            separator=" : "
            useTranslate3d={false}
            viewBox={
              Object {
                "height": 0,
                "width": 0,
                "x": 0,
                "y": 0,
              }
            }
            wrapperStyle={Object {}}
          />,
          <Legend
            align="center"
            iconSize={14}
            layout="horizontal"
            verticalAlign="bottom"
          />,
          <Line
            activeDot={true}
            animateNewValues={true}
            animationBegin={0}
            animationDuration={1500}
            animationEasing="ease"
            connectNulls={false}
            dataKey="value"
            dot={true}
            fill="#fff"
            hide={false}
            isAnimationActive={true}
            legendType="line"
            name="Power consumption"
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            points={Array []}
            stroke="#3182bd"
            strokeWidth={1}
            type="monotone"
            xAxisId={0}
            yAxisId={0}
          />,
        ],
        "data": Array [
          Object {
            "label": "15:40:47",
            "value": 265,
          },
        ],
        "height": 350,
        "layout": "horizontal",
        "margin": Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        },
        "reverseStackOrder": false,
        "stackOffset": "none",
        "width": 1200,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fill": "none",
            "horizontal": true,
            "horizontalFill": Array [],
            "horizontalPoints": Array [],
            "stroke": "#ccc",
            "vertical": true,
            "verticalFill": Array [],
            "verticalPoints": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "dataKey": "label",
            "domain": Array [
              0,
              "auto",
            ],
            "height": 30,
            "hide": false,
            "mirror": false,
            "orientation": "bottom",
            "padding": Object {
              "left": 0,
              "right": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "category",
            "width": 0,
            "xAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "domain": Array [
              0,
              "auto",
            ],
            "height": 0,
            "hide": false,
            "interval": 0,
            "mirror": false,
            "orientation": "left",
            "padding": Object {
              "bottom": 0,
              "top": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "number",
            "unit": " W",
            "width": 60,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "animationDuration": 400,
            "animationEasing": "ease",
            "contentStyle": Object {},
            "coordinate": Object {
              "x": 0,
              "y": 0,
            },
            "cursor": true,
            "cursorStyle": Object {},
            "filterNull": true,
            "isAnimationActive": true,
            "itemSorter": [Function],
            "itemStyle": Object {},
            "labelStyle": Object {},
            "offset": 10,
            "separator": " : ",
            "useTranslate3d": false,
            "viewBox": Object {
              "height": 0,
              "width": 0,
              "x": 0,
              "y": 0,
            },
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "iconSize": 14,
            "layout": "horizontal",
            "verticalAlign": "bottom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeDot": true,
            "animateNewValues": true,
            "animationBegin": 0,
            "animationDuration": 1500,
            "animationEasing": "ease",
            "connectNulls": false,
            "dataKey": "value",
            "dot": true,
            "fill": "#fff",
            "hide": false,
            "isAnimationActive": true,
            "legendType": "line",
            "name": "Power consumption",
            "onAnimationEnd": [Function],
            "onAnimationStart": [Function],
            "points": Array [],
            "stroke": "#3182bd",
            "strokeWidth": 1,
            "type": "monotone",
            "xAxisId": 0,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
